File: backend/src/application/schemas/coordination.py
from pydantic import BaseModel
from typing import List, Dict, Any
from datetime import datetime

class CoordinationSessionDTO(BaseModel):
    session_id: str
    status: str
    active_agents: int
    started_at: datetime

class CoordinationConsensusDTO(BaseModel):
    decision_id: str
    topic: str
    resolution: str
    agents_involved: List[str]

class CoordinationHandoffDTO(BaseModel):
    handoff_id: str
    from_agent: str
    to_agent: str
    reason: str
    timestamp: datetime

class CoordinationConflictDTO(BaseModel):
    conflict_id: str
    issue: str
    status: str
    resolved_by: str | None

File: frontend/src/types/coordination.ts
import { StandardResponse } from "./dashboard";

export interface CoordinationSession {
  session_id: string;
  status: string;
  active_agents: number;
  started_at: string;
}

export interface CoordinationConsensus {
  decision_id: string;
  topic: string;
  resolution: string;
  agents_involved: string[];
}

export interface CoordinationHandoff {
  handoff_id: string;
  from_agent: string;
  to_agent: string;
  reason: string;
  timestamp: string;
}

export interface CoordinationConflict {
  conflict_id: string;
  issue: string;
  status: string;
  resolved_by: string | null;
}
